applications/GeoMechanicsApplication/custom_utilities/transport_equation_utilities.hpp

```diff
@@ -0,0 +1,68 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Ignasi de Pouplana,
+//                   Vahid Galavi
+//                   Gennady Markelov
+//
+
+#pragma once
+
+// Project includes
+
+// Application includes
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class GeoTransportEquationUtilities
+{
+public:
+    using VectorType = Vector;
+    using MatrixType = Matrix;
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    template <unsigned int TDim, unsigned int TNumNodes>
+    static inline void CalculatePermeabilityMatrixH(BoundedMatrix<double, TNumNodes, TNumNodes>& rPMatrix,
+                                                    const Matrix& GradNpT,
+                                                    double        DynamicViscosityInverse,
+                                                    const BoundedMatrix<double, TDim, TDim>& PermeabilityMatrix,
+                                                    double IntegrationCoefficient)
+    {
+        noalias(rPMatrix) = DynamicViscosityInverse *
+                            prod(Matrix(prod(GradNpT, PermeabilityMatrix)), trans(GradNpT)) * IntegrationCoefficient;
+    }
+
+    static inline Matrix CalculatePermeabilityMatrixH(const Matrix& GradNpT,
+                                                      double        DynamicViscosityInverse,
+                                                      const Matrix& PermeabilityMatrix,
+                                                      double        IntegrationCoefficient)
+    {
+        return DynamicViscosityInverse *
+               prod(GradNpT, Matrix(prod(PermeabilityMatrix, trans(GradNpT)))) * IntegrationCoefficient;
+    }
+
+    template <unsigned int TNumNodes>
+    static inline void PreparePermeabilityMatrixHForIntegration(BoundedMatrix<double, TNumNodes, TNumNodes>& rPMatrix,
```

Comment:
These variables can be moved back. As far as I remember than it will be not really the permeability matrix. As well it seems this function should have another name because it could be used for another matrices. Looking forward to comments from Wijtze-Pieter and Anne.

Reply:
The only thing this function and its overload do is to multiply a matrix by a scalar (which is the product of three other scalars). Therefore, extracting such a function only makes sense if the name of that function clarifies the meaning of the resulting matrix-scalar product. I'm not sure whether that applies here.